api/s2n.h

```diff
@@ -23,6 +23,11 @@ extern "C" {
 #include <stdint.h>
 #include <openssl/ossl_typ.h>
 
+/* Function return code  */
+#define S2N_SUCCESS 0
+#define S2N_FAILURE -1
+#define S2N_AGAIN 1
```

Comment:
We spoke a bit about this but I don't think we should do this :) It's too big a backwards incompatible change for now. Instead I think we should return -1 and expect callers to do what they do with non-blocking I/O in general.

Reply:
Yes. I agree. I am removing S2N_AGAIN from the code and use S2N_ERR_BLOCKED to mark blocked request.